src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -46,6 +46,17 @@ public static IOperation GetRootOperation(this ISymbol symbol, CancellationToken
             }
         }
 
+        public static IEnumerable<ILocalSymbol> GetDeclaredVariables(this IVariableDeclarationStatement declarationStatement)
+        {
+            foreach (IVariableDeclaration group in declarationStatement.Declarations)
```

Comment:
linq is undesirable for the allocations it can cause.  However, you're not making things better since you're using yield here :)  You are effectively using linq.   You're just unwinding a select-many 'in place'.  

I'd be ok with an approach that just used an ArrayBuilder, did two foreachs that placed the variables in that, and then returned ToImmutableAndFree.  But using 'yield' as a half-measure to avoid linq doesn't really do anything.